source/Planet.cpp

```diff
@@ -472,10 +473,23 @@ void Planet::Bribe(bool fullAccess) const
 // Demand tribute, and get the planet's response.
 string Planet::DemandTribute(PlayerInfo &player) const
 {
+	map<string, string> subs;
+	subs["<tribute>"] = Format::Number(tribute) + " credits";
+	subs["<first>"] = player.FirstName();
+	subs["<last>"] = player.LastName();
+	subs["<origin>"] = Name();
+	subs["<ship>"] = player.Flagship()->Name();
+	static const string tributeHail = "";
 	if(player.GetCondition("tribute: " + name))
-		return "We are already paying you as much as we can afford.";
+	{
+		tributeHail = GetGovernment()->GetTributeHail(0);
```

Comment:
MZ generally defines a constant value and then refers to that constant, rather than the value of the constant, to improve code readability.
So instead of `(0)`, you'd do `(TRIBUTE_PRESENT)` here, and before your `subs` declaration you'd declare the values of your constants:
```
static const int TRIBUTE_PRESENT = 0x01
static const int TRIBUTE_IGNORED = 0x02
...
```